### PR TITLE
Try to fix issue265. When 'store' is filtered, resolve agagin

### PR DESCRIPTION
**Pull request by zeminzhou**
Try to fix [issue265](https://github.com/tikv/client-go/issues/265).
When the 'store' is filtered, save to 'filteredStore'. At the end of function, call 'reResolve' function for 'store' which save to 'filteredStore'.

### CODE REVIEW

**Comment by zeminzhou** [comment]
@youjiali1995 PTAL

File: internal/locate/region_cache.go [hunk 1]
```diff
@@ -172,6 +174,14 @@ func (r *regionStore) clone() *regionStore {
 
 // return next follower store's index
 func (r *regionStore) follower(seed uint32, op *storeSelectorOp) AccessIndex {
+	filteredStore := []*Store{}
+	defer func(filteredStore []*Store) {
+		for _, s := range filteredStore {
+			_, err := s.reResolve(r.regionCache)
+			tikverr.Log(err)
+		}
+	}(filteredStore)
```
Comment (disksing) [inline internal/locate/region_cache.go:183]:
```suggestion
	var filteredStore []*Store
	defer func() {
		for _, s := range filteredStore {
			_, err := s.reResolve(r.regionCache)
			tikverr.Log(err)
		}
	}()
```

**Comment by sticnarf** [comment]
It seems the `follower` method can be only called from the legacy `GetTiKVRPCContext`. But we have switched to `replicaSelector` for all TiKV requests. `GetTiKVRPCContext` is completedly unused now. cc @youjiali1995

**Comment by zeminzhou** [comment]
> It seems the `follower` method can be only called from the legacy `GetTiKVRPCContext`. But we have switched to `replicaSelector` for all TiKV requests. `GetTiKVRPCContext` is completedly unused now. cc @youjiali1995
> 
> So, I think to improve the similar case in the new implementation, we need to re-resolve the followers skipped by `accessFollower`?

**Comment by zeminzhou** [comment]
> It seems the `follower` method can be only called from the legacy `GetTiKVRPCContext`. But we have switched to `replicaSelector` for all TiKV requests. `GetTiKVRPCContext` is completedly unused now. cc @youjiali1995
> 
> So, I think to improve the similar case in the new implementation, we need to re-resolve the followers skipped by `accessFollower`?

New implementation means region_request.go ?

**Comment by sticnarf** [comment]
> New implementation means region_request.go ?

Yes, it's in `region_request.go`.

But it seems to me that the exact problem does not exist now... When the store epoch is updated, it will inform `asyncCheckAndResolveLoop` to update store info. @youjiali1995 can help confirm if I understand the original issue correctly...

**Review by youjiali1995 — COMMENTED** [review]
> But it seems to me that the exact problem does not exist now... When the store epoch is updated, it will inform asyncCheckAndResolveLoop to update store info. @youjiali1995 can help confirm if I understand the original issue correctly...

This issue doesn't exist now. 

And `reResolve` stores can't solve the issue in the old implementation. A peer can be accessed only if its `regionStore.storeEpochs` is equal to the corresponding `Store.epoch`. `regionStore.storeEpochs` won't be changed until a region is invalidated in the old implementation.

**Pull request closed without merging**